field.h

```diff
@@ -22,7 +25,9 @@ class GF2E {
   uint64_t data;
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wignored-attributes"
-  static std::function<uint64_t(__m128i)> reduce;
+  static uint64_t (*reduce_naive)(__m128i);
```

Comment:
This should probably be only a single function pointer that gets set to the fastest variant for each field size.